### DISCUSSION TITLE
StaticImage is not working inside array of image paths

### DISCUSSION THREAD

**Discussion by dohoangn** [gatsbyjs/gatsby#32427] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I have simple array of strings contain image names and try to use StaticImage to display image but not working
import React from "react";
import { Container } from "react-bootstrap";

import Layout from "../components/layout";
import { StaticImage } from "gatsby-plugin-image";

const carLogos = [
  "acura.gif",
  "alfa-romeo.gif"
];
const logo = "acura.gif";

const IndexPage = () => (  
  <Layout>
    <Container>     
        {carLogos.map((l) => (
          <p>
            <StaticImage src={"../images/car-logo/" + l} alt="Trust Us" />
          </p>
        ))}
    </Container>
  </Layout>
);

export default IndexPage;

However if I change  <StaticImage src={"../images/car-logo/" + logo} alt="Trust Us" /> which refer to variable, not array, then it is working

### Reproduction Link

https://donthaveone

### Steps to Reproduce

copy this IndexPage and run server

### Expected Result

Images should render

### Actual Result

Images aren't shown

### Environment

```shell
Windows, Mac
```

**Comment by LekoArts** [comment, marked as answer]
Please see https://github.com/gatsbyjs/gatsby/discussions/31760
